packages/ngtools/webpack/src/transformers/find_image_domains_spec.ts

```diff
@@ -0,0 +1,153 @@
+
+function findDomains(
+  input: string,
+  importHelpers = true,
+  module: ts.ModuleKind = ts.ModuleKind.ES2020,
+) {
+  const { program, compilerHost } = createTypescriptContext(input, undefined, undefined, {
+    importHelpers,
+    module,
+  });
+  const domains = new Set<string>();
+  const getTypeChecker = () => program.getTypeChecker();
+  const transformer = findImageDomains(getTypeChecker, domains);
+
+  transformTypescript(input, [transformer], program, compilerHost);
+
+  return domains;
+}
+
+function inputTemplate(provider: string) {
+  /* eslint-disable max-len */
+  return tags.stripIndent`
+    export class AppModule {
+        static ɵfac = function AppModule_Factory(t) { return new (t || AppModule)(); };
+        static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: AppModule, bootstrap: [AppComponent] });
+        static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [
+                ${provider}
+            ], imports: [BrowserModule] });
+    }
+    (function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(AppModule, [{
+            type: NgModule,
+            args: [{
+                    declarations: [
+                        AppComponent
+                    ],
+                    imports: [
+                        BrowserModule,
+                        NgOptimizedImage
+                    ],
+                    providers: [
+                        ${provider}
+                    ],
+                    bootstrap: [AppComponent]
+                }]
+        }], null, null); })();
+    (function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(AppModule, { declarations: [AppComponent], imports: [BrowserModule,
+            NgOptimizedImage] }); })();
+    `;
+}
+
+describe('@ngtools/webpack transformers', () => {
+  describe('find_image_domains', () => {
+    it('should find a domain when a built-in loader is used with a string-literal-like argument', () => {
+      // Intentionally inconsistent use of quote styles in this data structure:
+      const builtInLoaders: Array<[string, string]> = [
+        ['provideCloudflareLoader("www.cloudflareUrl.com")', 'www.cloudflareUrl.com'],
+        [
+          "provideCloudinaryLoader('https://www.cloudinaryUrl.net')",
+          'https://www.cloudinaryUrl.net',
+        ],
+        ['provideImageKitLoader("www.imageKitUrl.com")', 'www.imageKitUrl.com'],
+        ['provideImgixLoader(`www.imgixUrl.com/images/`)', 'www.imgixUrl.com/images/'],
+      ];
+      for (const loader of builtInLoaders) {
+        const input = inputTemplate(loader[0]);
+        const result = Array.from(findDomains(input));
+        expect(result.length).toBe(1);
+        expect(result[0]).toBe(loader[1]);
+      }
+    });
+
+    it('should not find a domain when a built-in loader is used with a variable', () => {
+      const input = inputTemplate(`provideCloudflareLoader(myImageCDN)`);
```

Comment:
Thought for docs: we should probably add a note that recommends using a raw string to allow preconnect generation to work